packages/opentelemetry-resources/src/platform/node/detectors/AwsEc2Detector.ts

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as http from 'http';
```

Comment:
https?

Reply:
can/should we use https for this?

Reply:
It appears as though http is the only option: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html